cmd/kubeadm/app/apis/output/v1alpha1/types.go

```diff
@@ -31,3 +31,13 @@ type BootstrapToken struct {
 
 	kubeadmapiv1beta2.BootstrapToken
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Image represents information for the output produced by 'kubeadm config images list'
+type Image struct {
+	metav1.TypeMeta `json:",inline"`
+
+	Name    string `json:"name"`
+	Version string `json:"version"`
+}
```

Comment:
this opens the same discussion as the `upgrade plan` PR:
streaming vs object separation

my vote is for the command to have the parent `Images` structure:
```
type Images struct {
	metav1.TypeMeta `json:",inline"`

	Images []Image `json:"images"`
}

type Image struct {
	Name    string `json:"name"`
	Version string `json:"version"`
}
```

this leaves us the option to add extra contents to the Images structure in the future.
streaming is not really required for this command, and for the time being.

i'm sure @rosti will agree to this.

Reply:
Absolutely, that would make parsing a lot easier. In the JSON case, we end up with a bunch of objects, that are not separated in any way. Making it hard for users to parse them. In addition to that, image listing is an operation that takes a negligibly small amount of time to complete. Hence, no buffering and separate objects are needed.

Reply:
@neolit123 @rosti Thank you for the review. Looks like I should also get rid of unbuffered output in the current implementation of [token output](https://github.com/kubernetes/kubernetes/pull/78764). Please, confirm.

Reply:
does the token output looks like this?

```
BootstrapToken
---
BootstrapToken
---
```

i'm sorry but i forgot what we are doing there. it makes sense if we are fetching one token at a time, if we are fetching them all at once it makes sense to have a parent structure:

```
BootstrapTokens {
    .... []BootstrapToken
}
```

the parent structure is preferred in all cases, unless we want it unbuffered - e.g. making network calls that are slow.

Reply:
/cc @akutz 
do you have an opinion on this topic?